File: Fortran/007.f90
program leetcode07
    implicit none
    integer(kind=4) :: a=89321
    call reverse_int(a)
    write(*,*) 'The reverse number is', a
end program leetcode07

subroutine reverse_int(a)
    implicit none
    integer(kind=4) :: a
    integer(kind=4) :: temp=0
    integer(kind=4) :: tx
    tx = abs(a)
    do while(abs(tx)>0)
        temp = temp*10+mod(tx,10)
        tx=tx/10
    end do

    if(temp>2147483647 .or. temp<-2147483647) then
        a=0
    else if (a>0) then
        a=temp
    else
        a=-temp
    end if

    return
end subroutine reverse_int
!gfortran 004.f90
!./a.out
!output 12398

File: Python/007.py
class Solution007(object):
    def reverse(self, x) :
        if x>=0:
            x = int(str(x)[::-1])
        else:
            x = -int(str(-x)[::-1])
        if x < 2147483647 and x >= -2147483648: #in the range of [-2^31,2^31-1]
            return x
        else:
            return 0
 
 
test07=Solution007()
test07.reverse(321)
#output 321
